src/Slices/InstanceComposerEditor/Core/helper.ts

```diff
@@ -0,0 +1,26 @@
+import { ServiceModel } from "@/Core";
+
+/**
+ * Recursively filters an array of ServiceModel objects based on possible ownership of the root instance.
+ *
+ * @param {ServiceModel[]} services - An array of ServiceModel objects to filter.
+ * @param {ServiceModel} mainService - root service to filter owned services by it.
+ * @returns {ServiceModel[]} An array of owned ServiceModel objects
+ */
+export const filterServices = (
+  services: ServiceModel[],
+  mainService: ServiceModel,
+): ServiceModel[] => {
+  const relatedServices = services.filter((service) =>
+    mainService?.owned_entities.includes(service.name),
+  );
+  const unrelatedServices = services.filter(
+    (service) => !mainService?.owned_entities.includes(service.name),
```

Comment:
Same observation here, the fact you are needing to use the optional chaining operator is Typsecript telling you there is a loophole where the value could potentially be null or undefined. 

I wouldn't mind it if it was used in the template (most often Angular practice I believe) but here, you're chaining a function to it that could cause a runtime error

`(service) =>
      !Array.isArray(mainService.owned_entities) || !mainService.owned_entities.includes(service.name),`